### YamlMime:JavaType
uid: "com.azure.core.exception.HttpResponseException"
fullName: "com.azure.core.exception.HttpResponseException"
name: "HttpResponseException"
nameWithType: "HttpResponseException"
summary: "The exception thrown when an unsuccessful response is received with http status code (e.<wbr>g."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Throwable?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Exception?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.RuntimeException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.AzureException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedClassMethods:
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#clone--\">clone</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-\">equals</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--\">finalize</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#getClass--\">getClass</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--\">hashCode</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notify--\">notify</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notifyAll--\">notifyAll</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-int-\">wait</a>"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#addSuppressed-java.lang.Throwable-\">addSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#fillInStackTrace--\">fillInStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getCause--\">getCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getLocalizedMessage--\">getLocalizedMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getMessage--\">getMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getStackTrace--\">getStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getSuppressed--\">getSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#initCause-java.lang.Throwable-\">initCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace--\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintStream-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintWriter-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#setStackTrace-java.lang.StackTraceElement[]-\">setStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#toString--\">toString</a>"
syntax: "public class **HttpResponseException**</br> extends <xref href=\"com.azure.core.exception.AzureException?alt=com.azure.core.exception.AzureException&text=AzureException\" data-throw-if-not-resolved=\"False\" />"
constructors:
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(com.azure.core.http.HttpResponse)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(HttpResponse response)"
  name: "HttpResponseException(HttpResponse response)"
  nameWithType: "HttpResponseException.HttpResponseException(HttpResponse response)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public HttpResponseException(HttpResponse response)"
  desc: "Initializes a new instance of the HttpResponseException class."
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(com.azure.core.http.HttpResponse,java.lang.Throwable)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(HttpResponse response, Throwable cause)"
  name: "HttpResponseException(HttpResponse response, Throwable cause)"
  nameWithType: "HttpResponseException.HttpResponseException(HttpResponse response, Throwable cause)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The <xref uid=\"java.lang.Throwable\" data-throw-if-not-resolved=\"false\" data-raw-source=\"Throwable\"></xref> which caused the creation of this exception."
    name: "cause"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  syntax: "public HttpResponseException(HttpResponse response, Throwable cause)"
  desc: "Initializes a new instance of the HttpResponseException class."
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(java.lang.String,com.azure.core.http.HttpResponse)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(String message, HttpResponse response)"
  name: "HttpResponseException(String message, HttpResponse response)"
  nameWithType: "HttpResponseException.HttpResponseException(String message, HttpResponse response)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The exception message."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public HttpResponseException(String message, HttpResponse response)"
  desc: "Initializes a new instance of the HttpResponseException class."
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Object)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(String message, HttpResponse response, Object value)"
  name: "HttpResponseException(String message, HttpResponse response, Object value)"
  nameWithType: "HttpResponseException.HttpResponseException(String message, HttpResponse response, Object value)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The exception message."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The deserialized response value."
    name: "value"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  syntax: "public HttpResponseException(String message, HttpResponse response, Object value)"
  desc: "Initializes a new instance of the HttpResponseException class."
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Throwable)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(String message, HttpResponse response, Throwable cause)"
  name: "HttpResponseException(String message, HttpResponse response, Throwable cause)"
  nameWithType: "HttpResponseException.HttpResponseException(String message, HttpResponse response, Throwable cause)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The exception message."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The <xref uid=\"java.lang.Throwable\" data-throw-if-not-resolved=\"false\" data-raw-source=\"Throwable\"></xref> which caused the creation of this exception."
    name: "cause"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  syntax: "public HttpResponseException(String message, HttpResponse response, Throwable cause)"
  desc: "Initializes a new instance of the HttpResponseException class."
- uid: "com.azure.core.exception.HttpResponseException.HttpResponseException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Throwable,boolean,boolean)"
  fullName: "com.azure.core.exception.HttpResponseException.HttpResponseException(String message, HttpResponse response, Throwable cause, boolean enableSuppression, boolean writableStackTrace)"
  name: "HttpResponseException(String message, HttpResponse response, Throwable cause, boolean enableSuppression, boolean writableStackTrace)"
  nameWithType: "HttpResponseException.HttpResponseException(String message, HttpResponse response, Throwable cause, boolean enableSuppression, boolean writableStackTrace)"
  summary: "Initializes a new instance of the Http<wbr>Response<wbr>Exception class."
  parameters:
  - description: "The exception message."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The <xref uid=\"java.lang.Throwable\" data-throw-if-not-resolved=\"false\" data-raw-source=\"Throwable\"></xref> which caused the creation of this exception."
    name: "cause"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  - description: "Whether suppression is enabled or disabled."
    name: "enableSuppression"
    type: "<xref href=\"boolean?alt=boolean&text=boolean\" data-throw-if-not-resolved=\"False\" />"
  - description: "Whether the exception stack trace will be filled in."
    name: "writableStackTrace"
    type: "<xref href=\"boolean?alt=boolean&text=boolean\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public HttpResponseException(String message, HttpResponse response, Throwable cause, boolean enableSuppression, boolean writableStackTrace)"
  desc: "Initializes a new instance of the HttpResponseException class."
methods:
- uid: "com.azure.core.exception.HttpResponseException.getResponse()"
  fullName: "com.azure.core.exception.HttpResponseException.getResponse()"
  name: "getResponse()"
  nameWithType: "HttpResponseException.getResponse()"
  summary: "Gets the <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
  syntax: "public HttpResponse getResponse()"
  desc: "Gets the <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
  returns:
    description: "The <xref uid=\"com.azure.core.http.HttpResponse\" data-throw-if-not-resolved=\"false\" data-raw-source=\"HttpResponse\"></xref> received that is associated to the exception."
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
- uid: "com.azure.core.exception.HttpResponseException.getValue()"
  fullName: "com.azure.core.exception.HttpResponseException.getValue()"
  name: "getValue()"
  nameWithType: "HttpResponseException.getValue()"
  summary: "Gets the deserialized HTTP response value."
  syntax: "public Object getValue()"
  desc: "Gets the deserialized HTTP response value."
  returns:
    description: "The deserialized HTTP response value."
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
type: "class"
desc: "The exception thrown when an unsuccessful response is received with http status code (e.g. 3XX, 4XX, 5XX) from the service request."
metadata: {}
package: "com.azure.core.exception"
artifact: com.azure:azure-core:1.34.0
